scripts/license_scanner/license_scanner.py

```diff
@@ -0,0 +1,128 @@
+        """Scan directory tree for filenames matching license_regex.
+
+        :param path: Path of the directory to run scanner
+        :return: Package paths and their corresponding license file contents
+        :rtype: dict
+        """
+        licenses = 0
+        license_files = {}
+
+        for dirpath, dirnames, filenames in os.walk(path):
+            for file in filenames:
+                if self.license_regex.match(file):
+                    license_file_content = self._get_license_file_contents(os.path.join(dirpath, file))
+                    rel_dirpath = os.path.relpath(dirpath, path)  # Limit path inside scanned directory
+                    license_files[rel_dirpath] = license_file_content
+                    licenses += 1
+                    print(f'License file: {os.path.join(dirpath, file)}')
+
+            # Remove directories that should not be scanned
+            for dir in self.config_data['excluded_directories']:
+                if dir in dirnames:
+                    dirnames.remove(dir)
+        print(f'{licenses} license files found.')
+        return license_files
+
+    def _get_license_file_contents(self, filepath):
+        try:
+            with open(filepath, encoding='utf8') as f:
+                return f.read()
+        except UnicodeDecodeError:
+            print(f'Unable to read license file: {filepath}')
+            pass
+
+    def create_license_file(self, licenses, filepath='NOTICES.txt'):
+        """Creates file with all the provided license file contents.
+
+        :param licenses: Dict with package paths and their corresponding license file contents
+        :param filepath: Path to write the file
+       """
+        package_separator = '------------------------------------'
+        with open(filepath, 'w', encoding='utf8') as f:
+            for directory, license in licenses.items():
+                license_output = '\n\n'.join([
+                    f'{package_separator}',
+                    f'Package path: {directory}',
+                    'License:',
+                    f'{license}\n'
+                ])
+                f.write(license_output)
+        return None
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description='Script to run LicenseScanner and generate license file')
+    parser.add_argument('--config-file', '-c', help='Config file for LicenseScanner')
+    parser.add_argument('--license-file-path', '-l', help='Create license file in the provided path')
+    parser.add_argument('--scan-path', '-s', default=os.curdir, help='Path to scan')
+    return parser.parse_args()
+
+
+def main():
+    try:
+        args = parse_args()
+        ls = LicenseScanner(args.config_file)
+        licenses = ls.scan(args.scan_path)
+
+        if args.license_file_path:
+            ls.create_license_file(licenses, args.license_file_path)
+    except Exception as e:
+        print(f'Type: {type(e).__name__}, Error: {e}')
+        return 1
```

Comment:
It is not a good practice to globally catch an exceptions that are known about it.
Every python method that is invoked in this file has a specific set of exceptions that can occur and therefore the exception handling should be constrained to that level.